week3/homework/src/actions.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const fs = require('fs');
+
+const ENCODING = 'utf8';
+
+class Actions {
+  constructor(filename) {
+    this._filename = filename;
+  }
+
+  list() {
+    return new Promise((resolve, reject) => {
+      fs.readFile(this._filename, ENCODING, (err, todoList) => {
+        if (err) {
+          if (err.code === 'ENOENT') console.log('no data found');
+          reject(err);
+        }
+        resolve(todoList);
+      });
+    });
+  }
+  reset(data) {
+    return new Promise((resolve, reject) => {
+      fs.writeFile(this._filename, JSON.stringify(data), err => {
+        if (err) {
+          reject(err);
+        }
+        resolve();
+      });
+    });
+  }
+  async update(request, response, value) {
```

Comment:
We should be able to use this function in any environment; in web server, CLI or something else. That means it should receive the data in a parameter and returns back the results or error as fulfilled or rejected promise (exactly like what you did in `list` and `reset` functions). It is not the responsibility of this function to read from the HTTP request or send a response.